### PR TITLE
Common: Remove TLS usage in GetLastErrorMsg

### PR DESCRIPTION
**Pull request by degasus**
Just return std::string as this code isn't performance critical.

### CODE REVIEW

File: Source/Core/Common/Misc.cpp [hunk 1]
```diff
@@ -8,27 +8,19 @@
...
-
 // Generic function to get last error message.
 // Call directly after the command or use the error num.
 // This function might change the error code.
-const char* GetLastErrorMsg()
+std::string GetLastErrorMsg()
 {
-	static const size_t buff_size = 255;
+	const size_t buff_size = 255;
+	char err_str[buff_size] = {};
```
Comment (degasus) [inline Source/Core/Common/Misc.cpp]: <Alcaro> huh. why initialize that char buffer? That just looks like a waste of cpu to me.

Comment (lioncash) [inline Source/Core/Common/Misc.cpp]: Realistically, if initializing a 255 char buffer brings your CPU to it's knees, you have other problems.

**Comment by lioncash** [comment]
Looks OK to me.

**Comment by Parlane** [comment]
LGTM

**Pull request merged**